modules/programs/waybar.nix

```diff
@@ -343,9 +353,9 @@ in {
           Description =
             "Highly customizable Wayland bar for Sway and Wlroots based compositors.";
           Documentation = "https://github.com/Alexays/Waybar/wiki";
-          PartOf = [ "graphical-session.target" ];
+          PartOf = [ cfg.systemd.target ];
           Requisite = [ "dbus.service" ];
-          After = [ "dbus.service" ];
+          After = [ cfg.systemd.target ];
```

Comment:
The service type is set to `dbus` so it needs to be started only after dbus is started.
`After` makes sure the specified services are started if not active while `Requisite` imposes a dependency without starting the service.
The service is automatically started with the `WantedBy` directive below.

I would leave it as it is:
```
Requisite = [ "dbus.service" ];
After = [ "dbus.service" ];
```

See https://www.freedesktop.org/software/systemd/man/systemd.unit.html for a complete explanation.